pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/RoundDecimalTransformFunction.java

```diff
@@ -0,0 +1,100 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.operator.transform.function;
+
+import com.google.common.base.Preconditions;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.List;
+import java.util.Map;
+import org.apache.pinot.core.operator.blocks.ProjectionBlock;
+import org.apache.pinot.core.operator.transform.TransformResultMetadata;
+import org.apache.pinot.core.plan.DocIdSetPlanNode;
+import org.apache.pinot.segment.spi.datasource.DataSource;
+import org.apache.pinot.spi.data.FieldSpec;
+
+
+//TODO: The function should ideally be named 'round'
+// but it is not possible because of existing DateTimeFunction with same name.
+public class RoundDecimalTransformFunction extends BaseTransformFunction {
+  public static final String FUNCTION_NAME = "roundDecimal";
+  private double[] _result;
+  private TransformFunction _leftTransformFunction;
+  private TransformFunction _rightTransformFunction;
+
+  @Override
+  public String getName() {
+    return FUNCTION_NAME;
+  }
+
+  @Override
+  public void init(List<TransformFunction> arguments, Map<String, DataSource> dataSourceMap) {
+    int numArguments = arguments.size();
+    // Check that there are more than 2 arguments or no arguments
+    if (numArguments < 1 || numArguments > 2) {
+      throw new IllegalArgumentException(
+          "roundDecimal transform function supports either 1 or 2 arguments. Num arguments provided: " + numArguments);
+    }
+
+    _leftTransformFunction = arguments.get(0);
+    if (numArguments > 1) {
+      _rightTransformFunction = arguments.get(1);
+      Preconditions.checkArgument(_rightTransformFunction.getResultMetadata().isSingleValue()
+              && isArgumentDataTypeValid(_rightTransformFunction),
+          "Argument must be single-valued with type INT for transform function: %s", getName());
+    } else {
+      _rightTransformFunction = null;
+    }
+
+    Preconditions.checkArgument(_leftTransformFunction.getResultMetadata().isSingleValue(),
+        "Argument must be single-valued for transform function: %s", getName());
+  }
+
+  private boolean isArgumentDataTypeValid(TransformFunction transformFunction) {
+    return transformFunction.getResultMetadata().getDataType().getStoredType() == FieldSpec.DataType.INT
+        || transformFunction.getResultMetadata().getDataType().getStoredType() == FieldSpec.DataType.LONG;
```

Comment:
Suggesting adding a flag to `DataType` - `isIntegral` or something like that. It will be useful.

Reply:
done